src/itemlist_formaction.cpp

```diff
@@ -84,11 +84,19 @@ void itemlist_formaction::process_operation(operation op, bool automatic, std::v
 		LOG(level::INFO, "itemlist_formaction: opening item at pos `%s'", itemposname);
 		if (itemposname.length() > 0 && visible_items.size() != 0) {
 			if (itempos < visible_items.size()) {
-				visible_items[itempos].first->set_unread(false);
-				v->get_ctrl()->mark_article_read(visible_items[itempos].first->guid(), true);
-				v->open_in_browser(visible_items[itempos].first->link());
+				// store if open_in_browser() succeded
+				bool browser_success = v->open_in_browser(visible_items[itempos].first->link());
+				// do only mark as unread if open_in_browser() succeded
+				if (browser_success) {
+					visible_items[itempos].first->set_unread(false);
+					v->get_ctrl()->mark_article_read(visible_items[itempos].first->guid(), true);
+				}
```

Comment:
This code has two problems:
1. no error is displayed if browser fails;
2. it carries on to `OP_NEXTUNREAD` regardless of browser's exit code.

I think that instead of handling the happy case (`if (browser_success) …`), this should handle the failure (`if (! browser_success) …`). it could just show an error (grep for `show_error`) and `break` from this whole `switch` (or maybe `return` from the function; you decide).

This also means you won't have to add `browser_success` to the condition below.